core/include/detray/materials/material_rod.hpp

```diff
@@ -9,45 +9,67 @@
 
 // Project include(s)
 #include "detray/definitions/qualifiers.hpp"
+#include "detray/intersection/intersection.hpp"
 #include "detray/materials/material.hpp"
 
 namespace detray {
 
-// Rod structure to be mapped on the mask
+// Rod structure to be mapped on the line mask
 template <typename scalar_t>
 struct material_rod {
     using scalar_type = scalar_t;
     using material_type = material<scalar_t>;
 
     material_rod() = default;
 
-    material_rod(const material_type& material, scalar_type outer_r,
-                 scalar_type inner_r = 0)
-        : m_material(material), m_outer_r(outer_r), m_inner_r(inner_r) {}
+    material_rod(const material_type& material, scalar_type radius)
+        : m_material(material),
+          m_radius(radius),
+          m_radius_in_X0(radius / material.X0()),
+          m_radius_in_L0(radius / material.L0()) {}
 
     /// Equality operator
     ///
     /// @param rhs is the right hand side to be compared to
     DETRAY_HOST_DEVICE
     bool operator==(const material_rod<scalar_t>& rhs) const {
-        return (m_material == rhs.get_material() &&
-                m_outer_r == rhs.outer_r() && m_inner_r == rhs.inner_r());
+        return (m_material == rhs.get_material() && m_radius == rhs.radius());
     }
 
     /// Access the (average) material parameters.
     DETRAY_HOST_DEVICE
     constexpr const material_type& get_material() const { return m_material; }
-    /// Return the outer radius
+
+    /// Return the radius
     DETRAY_HOST_DEVICE
-    constexpr scalar_type outer_r() const { return m_outer_r; }
-    /// Return the inner radius
+    constexpr scalar_type radius() const { return m_radius; }
+
+    /// Return the interaction length
     DETRAY_HOST_DEVICE
-    constexpr scalar_type inner_r() const { return m_inner_r; }
+    scalar_type interaction_length(const line_plane_intersection& is) const {
+        // Assume that is.p2[0] is radial distance of line intersector
+        if (is.p2[0] > m_radius) {
```

Comment:
I guess the rod radius can be different from the mask radius due to material mapping? Probably also just an approximation for a line with square cross section? Sorry, I don't know much about it :D

Reply:
rod radius: physical radius of the line detector
mask radius: non-physical radius of cross secsion

The material interaction is only calculated when p2[0] (the distance of the closest approach) is smaller than the physical rod radius. 
Actually, we also need to take into account the gas material surrounding the wires, which is not considered in this PR.
There are couple of ways to handle the gas material but let me work it out later

Reply:
Yes we should. Wire is still in rod shape physically. (cross section of line mask and wire shape are independent to each other)